plugins/jobs/src/js/components/Breadcrumbs.js

```diff
@@ -6,57 +6,66 @@ import Breadcrumb from "#SRC/js/components/Breadcrumb";
 import BreadcrumbTextContent from "#SRC/js/components/BreadcrumbTextContent";
 import PageHeaderBreadcrumbs from "#SRC/js/components/PageHeaderBreadcrumbs";
 
-export default function Breadcrumbs({ item, children, states }) {
-  function getBreadcrumb(item, id = "", name = "") {
-    const isDetailPage = [...item.path, item.name].join(".") === id;
-    const link = isDetailPage ? `/jobs/detail/${id}` : `/jobs/overview/${id}`;
-
-    return (
-      <Breadcrumb key={id} title="Jobs">
-        <BreadcrumbTextContent>
-          <Link to={link}>{name === "" ? "Jobs" : name}</Link>
-        </BreadcrumbTextContent>
-
-        {isDetailPage ? states : null}
-      </Breadcrumb>
-    );
-  }
-
-  function getBreadcrumbList(item) {
-    if (item == null) {
-      return [];
-    }
+/**
+ * @param  {string} link
+ * @param  {string[]} pathParts path including name (if given)
+ * @param  {JSX.Element} [children=null] extra JSX Components to render
+ *
+ * @return {JSX.Element} Breadcrumb Part Component
+ */
+function BreadcrumbPart({ href, pathParts, children = null }) {
+  return (
+    <Breadcrumb key={pathParts.join(".")} title="Jobs">
+      <BreadcrumbTextContent>
+        <Link to={href}>{pathParts.slice(-1)}</Link>
+      </BreadcrumbTextContent>
+      {children}
+    </Breadcrumb>
+  );
+}
+BreadcrumbPart.propTypes = {
+  href: PropTypes.string.isRequired,
+  pathParts: PropTypes.arrayOf(PropTypes.string).isRequired
+};
 
-    const pathSegments = [...item.path, item.name];
-    const segments = pathSegments.map((_, index) =>
-      pathSegments.slice(0, index + 1)
-    );
+export default function Breadcrumbs({ path, name, children, jobInfo }) {
+  // we're starting from the back, so first we have children
+  let breadcrumbs = React.Children.toArray(children);
 
-    return segments.map(segment =>
-      getBreadcrumb(item, segment.join("."), segment[segment.length - 1])
-    );
+  // if we got a name, thats a detail link
+  if (name !== undefined) {
+    breadcrumbs = [
+      <BreadcrumbPart
+        href={["/jobs/detail/"].concat(path, [name]).join(".")}
```

Comment:
🎨 Can we move this into a separate testable / typeable function? This is too much logic inside a jsx assignment imho